plugins/querytranslate/search.go

```diff
@@ -64,17 +77,23 @@ func (query *Query) generateSearchQuery(rsQuery RSQuery) (*interface{}, error) {
 	return &searchQuery, nil
 }
 
-func (query *Query) generateShouldQuery() interface{} {
+func (query *Query) generateShouldQuery() (interface{}, error) {
 	var fields []string
 	var phrasePrefixFields []string
-	for index, field := range query.DataField {
+	normalizedFields := NormalizedDataFields(query.DataField, query.FieldWeights)
+	if len(normalizedFields) < 1 {
+		return nil, errors.New("Field 'dataField' cannot be empty")
+	}
+	for _, dataField := range normalizedFields {
 		var fieldWeight string
-		if len(query.FieldWeights) > index {
-			fieldWeight = strconv.FormatFloat(query.FieldWeights[index], 'f', 2, 64)
+		if dataField.Weight != 0 {
```

Comment:
```suggestion
		if dataField.Weight > 0 {
```

Reply:
Weights should always be positive integer or decimal points, can't be 0 or negative.